packages/client/src/utils.ts

```diff
@@ -0,0 +1,53 @@
+import { AES, enc, SHA256 } from 'crypto-js';
+
+export function encryptMessage(message: string, key: string): string {
+  return AES.encrypt(message, key).toString();
+}
+
+export function decryptMessage(encryptedMessage: string, key: string): string {
+  return AES.decrypt(encryptedMessage, key).toString(enc.Utf8);
+}
+
+export function generateRandomKey(): string {
+  const random = (Math.random() * 100000000).toString();
+  const date = new Date().toString();
+  return SHA256(random + date).toString();
+}
+
+export function storeKeyToLocalStorage(key: string): void {
+  localStorage.setItem('key', key);
+}
+
+export function getKeyFromLocalStorage(): string | null {
+  return localStorage.getItem('key');
+}
+
+export function getRawKeyFromLocalStorageIfPresentElseGenerate(
+  password?: string,
+): string {
+  const key = localStorage.getItem('key');
+  if (key !== null) {
+    if (password !== undefined) return decryptMessage(key, password);
+    return key;
+  }
+  const newKey = generateRandomKey();
+  if (password !== undefined) storeKeyToLocalStorage(encryptMessage(newKey, password));
+  else storeKeyToLocalStorage(newKey);
+
+  return newKey;
+}
```

Comment:
If there is no key stored in localStorage, a random key is generated and returned.